### PR TITLE
Disable STP processing on a specific port

### PR DESCRIPTION
**Pull request by mg964**
Add support for a "BPDU Port Filter" option ("setportbpdufilter"),
i.e. discard BPDUs on ingress and do not issue BPDUs. Further add a
counter that indicates the number of received BPDUs that have been
discarded.

### CODE REVIEW

**Comment by nihilus** [comment]
Actually counters is unnecessary overhead since this is being enforced by the user, and eventually they will wrap.

File: mstp.c [hunk 1]
```diff
@@ -3252,6 +3275,10 @@ static bool PTSM_run(port_t *prt, bool dry_run)
             }
             if(!(prt->txCount < prt->bridge->Transmit_Hold_Count))
                 return false;
+
+            if(prt->bpduFilterPort)
+                return false;
```
Comment (commodo) [inline mstp.c]: as a note, this patch here, differs a bit from the initial one, which was here: https://github.com/mstpd/mstpd/pull/61/commits/83eaaf81d74c954301a1df62429556d44be135d1#diff-afc426ea39d24015ac2bbcb1eed6a712L3256

**Comment by mg964** [comment]
"unnecessary overhead"? For me "bpdu-filter" is a bit of additional paranoia (along with bpdu-guard), in a well behaved network you would expect the counter to remain at 0. But if the counter is incrementing you may want to take further action - "why has this edge node started sending me BPDUs".

File: mstp.c (hunk 1 above)
Comment (mg964) [reply mstp.c]: Yeah, seemed prudent to perform the check before the send RSTP/STP check. Can move the check if desired.

Comment (commodo) [reply mstp.c]: No, that's fine.
Your patch makes a bit more sense than the initial one.

Comment (commodo) [reply mstp.c]: i.e. I understand it easier

**Comment by commodo** [comment]
The BPDU filter counters are a neat addition.
So, let's have them ; they can be useful when debugging the state-machine of STP.

**Comment by commodo** [comment]
[LGTM](https://i.pinimg.com/564x/80/dc/fe/80dcfe033af4ceea0c2611f3c9375b65.jpg)

**Comment by commodo** [comment]
Will merge this later today if no objections :)

**Comment by commodo** [comment]
Hmm, let's move this to tomorrow ; it gives other timezones a chance to look.
[ Sorry for the rush ]

**Comment by commodo** [comment]
@nihilus - are you referring to disabling just BPDU filter counters via configure-option ?

regarding "individual interfaces per port specific for STP (weird implementation)." - we also used something similar ; it's a pretty good way to use a switch ASIC to do switching and control bridging protocols (STP, LLDP) per-port with lldpd & mstpd ;

**Comment by nihilus** [comment]
@commodo Yes, exactly :-)

**Comment by commodo** [comment]
Well, I'm just trying to clarify what @nihilus is saying.

Personally, I'm not convinced that a `disabling just BPDU filter counters via configure-option` is a good idea. It's a bit like trying to use a sledge-hammer to kill a fly.
The overhead that they add is very minor, and a config option is the proverbial sledge-hammer.

I'd say, let's have this change as-is for now.
Maybe we can find a clearer reason later if we need something like this.

**Pull request merged**